packages/ra-ui-materialui/src/input/AutocompleteInput.tsx

```diff
@@ -421,30 +429,6 @@ If you provided a React element for the optionText prop, you must also provide t
 
     const oneSecondHasPassed = useTimeout(1000, filterValue);
 
-    // To avoid displaying an empty list of choices while a search is in progress,
-    // we store the last choices in a ref. We'll display those last choices until
-    // a second has passed.
-    const currentChoices = useRef(allChoices);
-
-    useEffect(() => {
-        if (allChoices && (allChoices.length > 0 || oneSecondHasPassed)) {
-            currentChoices.current = allChoices;
-        }
-    }, [allChoices, oneSecondHasPassed]);
-
-    const suggestions = useMemo(() => {
```

Comment:
We don't need to filter the suggestions ourselves: MUI's Autocomplete takes them all, but displays only the ones returned by `filterOptions`